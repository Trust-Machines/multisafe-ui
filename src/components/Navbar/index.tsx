import AppBar from '@mui/material/AppBar';
import Box from '@mui/material/Box';
import Toolbar from '@mui/material/Toolbar';
import Typography from '@mui/material/Typography';
import Button from '@mui/material/Button';
import MenuIcon from '@mui/icons-material/Menu';
import IconButton from '@mui/material/IconButton';
import LightModeIcon from '@mui/icons-material/LightMode';
import DarkModeIcon from '@mui/icons-material/DarkMode';

import useAppTheme from "../../hooks/useAppTheme";

const Navbar = () => {
    const [appTheme, toggleAppTheme] = useAppTheme();

    return (
        <Box sx={{flexGrow: 1}}>
            <AppBar position="static">
                <Toolbar>
                    <IconButton
                        size="large"
                        edge="start"
                        color="inherit"
                        aria-label="menu"
                        sx={{mr: 2}}
                    >
                        <MenuIcon/>
                    </IconButton>
                    <Typography variant="h6" component="div" sx={{flexGrow: 1}}>
                        MultiSafe
                    </Typography>
                    <Button color="inherit">Connect Wallet</Button>
                    <IconButton onClick={toggleAppTheme}>
                        {appTheme === "light" ? <DarkModeIcon/> : <LightModeIcon/>}
                    </IconButton>
                </Toolbar>
            </AppBar>
        </Box>
    );
}

export default Navbar;