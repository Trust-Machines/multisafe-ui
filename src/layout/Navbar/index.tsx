import {useState} from "react";
import AppBar from '@mui/material/AppBar';
import Toolbar from '@mui/material/Toolbar';
import Typography from '@mui/material/Typography';
import Button from '@mui/material/Button';
import IconButton from '@mui/material/IconButton';
import LightModeIcon from '@mui/icons-material/LightMode';
import DarkModeIcon from '@mui/icons-material/DarkMode';
import Container from '@mui/material/Container';
import Menu from '@mui/material/Menu';
import MenuItem from '@mui/material/MenuItem';
import Chip from '@mui/material/Chip';

import useAppTheme from "../../hooks/useAppTheme";
import useAuth from "../../hooks/useAuth";
import useNetwork from "../../hooks/useNetwork";

import NetworkDialog from "../../components/NetworkDialog";
import {Network} from "../../store/network";

const truncateMiddle = (s: string, cut: number): string => {
    const l = s.length;
    return s.substring(0, cut) + "..." + s.substring(l - cut, l);
}


const Navbar = () => {
    const [appTheme, toggleAppTheme] = useAppTheme();
    const [userData, openAuth, signOut] = useAuth();
    const [network, setNetwork] = useNetwork();
    const [anchorElMenu, setAnchorElMenu] = useState<null | HTMLElement>(null);
    const [networkDialog, setNetworkDialog] = useState(false);

    const handleOpenMenu = (event: React.MouseEvent<HTMLElement>) => {
        setAnchorElMenu(event.currentTarget);
    };

    const handleCloseMenu = () => {
        setAnchorElMenu(null);
    };

    const openNetworkDialog = () => {
        setNetworkDialog(true);
    }

    const closeNetworkDialog = (selected: Network) => {
        setNetworkDialog(false);

        if (selected !== network) {
            setNetwork(selected);
        }
    }

    return (
        <>
            <AppBar position="static">
                <Container maxWidth="lg">
                    <Toolbar>
                        <Typography variant="h6" component="div" sx={{flexGrow: 1}}>
                            MultiSafe
                        </Typography>
                        <Toolbar>
                            {userData &&
                            <Button variant="outlined" style={{background: appTheme === 'light' ? 'white' : 'inherit'}}
                                    onClick={handleOpenMenu}>{truncateMiddle(userData.profile.stxAddress[network], 4)}</Button>}
                            {!userData && <Button color="inherit" onClick={openAuth}>Connect Wallet</Button>}
                        </Toolbar>
                        <IconButton onClick={toggleAppTheme}>
                            {appTheme === "light" ? <DarkModeIcon/> : <LightModeIcon/>}
                        </IconButton>
                    </Toolbar>
                    <Menu
                        sx={{mt: '45px', minWidth: '400px'}}
                        id="menu-appbar"
                        anchorEl={anchorElMenu}
                        anchorOrigin={{
                            vertical: 'top',
                            horizontal: 'right',
                        }}
                        keepMounted
                        transformOrigin={{
                            vertical: 'top',
                            horizontal: 'right',
                        }}
                        open={Boolean(anchorElMenu)}
                        onClose={handleCloseMenu}
                    >
                        <MenuItem onClick={() => {
                            openNetworkDialog();
                            handleCloseMenu();
                        }}>
                            Network <Chip size="small" label={network} sx={{ml: '60px'}}/>
                        </MenuItem>
                        <MenuItem onClick={() => {
                            openAuth();
                            handleCloseMenu();
                        }}>
                            Switch Account
                        </MenuItem>
                        <MenuItem onClick={() => {
                            signOut();
                            handleCloseMenu();
                        }}>
                            Sign Out
                        </MenuItem>
                    </Menu>
                </Container>
            </AppBar>
            <NetworkDialog selectedValue={network} open={networkDialog} onClose={closeNetworkDialog}/>
        </>
    );
}

export default Navbar;