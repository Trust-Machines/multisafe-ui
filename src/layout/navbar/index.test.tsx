import {render} from '@testing-library/react';

import Navbar from './index';

test('1 Default render', () => {
    const rendered = render(<Navbar/>);
    expect(rendered.container).toMatchSnapshot()
});

