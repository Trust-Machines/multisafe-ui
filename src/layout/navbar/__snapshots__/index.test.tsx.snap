// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1 Default render 1`] = `
<div>
  <div
    class="MuiBox-root css-hoayk1"
  >
    <div
      class="MuiBox-root css-azxdeb"
    >
      <div
        class="MuiBox-root css-1qmgq4"
      >
        <div
          class="MuiBox-root css-wxdvxx"
        >
          <img
            alt="Logo"
            src="/logo.png"
          />
        </div>
        <div
          class="MuiBox-root css-1dx8jfp"
        >
          MultiSafe
        </div>
      </div>
    </div>
    <div
      class="flex-grow MuiBox-root css-0"
    />
    <div
      class="MuiBox-root css-ls0mhy"
    >
      <div
        class="MuiBox-root css-7lmw8c"
      >
        <div
          class="MuiBox-root css-198xhax"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-18wkfow-MuiSvgIcon-root"
            data-testid="AccountBalanceWalletIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M21 18v1c0 1.1-.9 2-2 2H5c-1.11 0-2-.9-2-2V5c0-1.1.89-2 2-2h14c1.1 0 2 .9 2 2v1h-9c-1.11 0-2 .9-2 2v8c0 1.1.89 2 2 2h9zm-9-2h10V8H12v8zm4-2.5c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5z"
            />
          </svg>
          <div
            class="MuiBox-root css-x1k79c"
          >
            <span
              class="first-line"
            >
              Not connected
            </span>
            <span
              class="second-line"
            >
              Connect Wallet
            </span>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-1n1j97o"
      >
        <div
          class="MuiBox-root css-1sfu4nm"
        >
          <div
            class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-clickable MuiChip-clickableColorPrimary MuiChip-filledPrimary css-8w4ikn-MuiButtonBase-root-MuiChip-root"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
            >
              mainnet
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-1qoaa23"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="DarkModeIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 3c-4.97 0-9 4.03-9 9s4.03 9 9 9 9-4.03 9-9c0-.46-.04-.92-.1-1.36-.98 1.37-2.58 2.26-4.4 2.26-2.98 0-5.4-2.42-5.4-5.4 0-1.81.89-3.42 2.26-4.4-.44-.06-.9-.1-1.36-.1z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
